src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs

```diff
@@ -304,20 +304,9 @@ private static void GenerateChildrenPut(NoxSolution solution, EntityRelationship
         var isSingleRelationship = relationship.WithSingleEntity();
         var relationshipName = parent.GetNavigationPropertyName(relationship);
 
-        if (isSingleRelationship)
-        {
-            code.AppendLine($"public virtual async Task<ActionResult<{child.Name}Dto>> PutTo{relationshipName}(" +
-                $"{GetPrimaryKeysRoute(parent, solution, attributePrefix: "")}, " +
-                $"[FromBody] {child.Name}UpsertDto {child.Name.ToLowerFirstChar()})");
-        }
-        else
-        {
-            code.AppendLine($"[HttpPut(\"{solution.Infrastructure.Endpoints.ApiRoutePrefix}/{parent.PluralName}/{PrimaryKeysAttribute(parent)}/{relationshipName}/{PrimaryKeysAttribute(child, "relatedKey")}\")]");
-            code.AppendLine($"public virtual async Task<ActionResult<{child.Name}Dto>> PutTo{child.PluralName}NonConventional(" +
-                $"{GetPrimaryKeysRoute(parent, solution, attributePrefix: "")}, " +
-                $"{GetPrimaryKeysRoute(child, solution, "relatedKey", "")}, " +
-                $"[FromBody] {child.Name}UpsertDto {child.Name.ToLowerFirstChar()})");
-        }
+        code.AppendLine($"public virtual async Task<ActionResult<{child.Name}Dto>> PutTo{relationshipName}(" +
```

Comment:
we should use the related entity name? to make it consistent with "normal" Relations
for owned we do not have the case of relating to the same entity

And we should keep the plural if ToMany and singular if to one?

We have other issue to rename the OwnedEntitites Navigation properties, and think we should align also the endpoints

Reply:
Another question, do you think its easy to migrate this to the template for owned, now that you are changing this?